extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -785,7 +785,7 @@ private void handleHibernateORMWithNoPersistenceXml(
                 }
 
                 hibernateConfig.database.charset.ifPresent(
-                        charset -> desc.getProperties().setProperty(AvailableSettings.HBM2DDL_CHARSET_NAME, charset));
+                        charset -> desc.getProperties().setProperty(AvailableSettings.HBM2DDL_CHARSET_NAME, charset.name()));
```

Comment:
So isn't it already implemented then? :]

(have a look at the name of the property)

Reply:
Oh man... I didn't even see that this was the same property... F**k

Reply:
Yeah...